ruby/BUILD.bazel

```diff
@@ -26,43 +26,31 @@ filegroup(
     name = "srcs",
     srcs = glob([
         "lib/**/*.rb",
-        "src/**/*.proto",
+        "ext/**/*.{c,h}}",
     ]) + [
         "Gemfile",
         "Rakefile",
         "google-protobuf.gemspec",
-        "pom.xml",
     ],
 )
 
 internal_ruby_extension(
     name = "protobuf_c_mac",
-    extension = "lib/google/protobuf_c.bundle",
+    extension = "ext/google/x86_64-darwin/libprotobuf_c.bundle",
     deps = glob(["ext/google/protobuf_c/*"]),
     target_compatible_with = select({
-        ":java_ruby": ["@platforms//:incompatible"],
+        "@platforms//os:linux": ["@platforms//:incompatible"],
         "//conditions:default": ["@platforms//os:osx"],
     }),
 )
 
 internal_ruby_extension(
     name = "protobuf_c",
-    extension = "lib/google/protobuf_c.so",
+    extension = "ext/google/x86_64-linux/libprotobuf_c.so",
```

Comment:
This will break on other architectures won't it?

Reply:
Yes it would, but this is one of only two architectures we use on CI. Per our chat earlier this week, this branch assumes that we're only solving for releasing source gem.

Reply:
Yea that makes sense, I guess my concern is for development and expansion of our cross-architecture CI builds.  We could always add more of these as needed though.

You should definitely mark these as only compatible with the specific architecture tho, so that wildcard bazel expansions don't pick them up on the wrong platform